src/components/PickerDay.vue

```diff
@@ -173,20 +192,68 @@ export default {
      */
     isLeftNavDisabled () {
       return this.isRtl
-        ? this.isNextMonthDisabled(this.pageTimestamp)
-        : this.isPreviousMonthDisabled(this.pageTimestamp)
+        ? this.isNextMonthDisabled()
+        : this.isPreviousMonthDisabled()
     },
     /**
      * Is the right hand navigation button disabled?
      * @return {Boolean}
      */
     isRightNavDisabled () {
       return this.isRtl
-        ? this.isPreviousMonthDisabled(this.pageTimestamp)
-        : this.isNextMonthDisabled(this.pageTimestamp)
+        ? this.isPreviousMonthDisabled()
+        : this.isNextMonthDisabled()
     }
   },
   methods: {
+    focusNextDay() {
+      const newDate = new Date(this.focusedDate);
+      newDate.setDate(newDate.getDate() + 1);
+
+      const monthsDifference = newDate.getMonth() - new Date(this.pageTimestamp).getMonth();
+      const shouldChangePage = this.sideBySide ? monthsDifference > 1 : monthsDifference > 0;
+      if (shouldChangePage) {
+        this.changeMonth(1);
+      }
+
+      this.$emit('update:focusedDate', newDate.getTime());
+    },
+    focusPreviousDay () {
+      const newDate = new Date(this.focusedDate);
+      newDate.setDate(newDate.getDate() - 1);
+
+      const monthsDifference = new Date(this.pageTimestamp).getMonth() - newDate.getMonth();
+      const shouldChangePage = monthsDifference > 0;
+      if (shouldChangePage) {
+        this.changeMonth(-1);
+      }
+
+      this.$emit('update:focusedDate', newDate.getTime());
+    },
+    focusNextWeek() {
+      const newDate = new Date(this.focusedDate);
+      newDate.setDate(newDate.getDate() + 7);
+
+      const monthsDifference = newDate.getMonth() - new Date(this.pageTimestamp).getMonth();
+      const shouldChangePage = this.sideBySide ? monthsDifference > 1 : monthsDifference > 0;
+      if (shouldChangePage) {
+        this.changeMonth(1);
+      }
+
+      this.$emit('update:focusedDate', newDate.getTime());
```

Comment:
This is repeated from `focusNextDay`. Let's keep it DRY.

Reply:
I've refactored this and caught a bug (I hadn't taken into account jumping to a different year)